File: Tests/SignalR.Extras.Autofac.Test/stubs/Dependencies/ObjectStubBase.fs
namespace SignalR.Extras.Autofac.Test.Stubs.Dependencies

open System

[<AbstractClass>]
type ObjectStubBase() =
    let onDisposing = Event<_>()
    member this.OnDisposing = onDisposing.Publish

    interface IDisposable with
        member this.Dispose() = onDisposing.Trigger()

File: Tests/SignalR.Extras.Autofac.Tests/Stubs/Dependencies/ObjectStubBase.cs
namespace SignalR.Extras.Autofac.Tests.Stubs.Dependencies;

public abstract class ObjectStubBase : IDisposable
{

    public event EventHandler OnDisposing;

    public void Dispose()
    {
        var handler = OnDisposing;
        handler?.Invoke(this, EventArgs.Empty);
    }

}